src/open_inwoner/kvk/client.py

```diff
@@ -101,10 +101,23 @@ def get_company_headquarters(self, kvk: str, **kwargs) -> dict:
 
     def get_all_company_branches(self, kvk: str, **kwargs) -> list[Optional[dict]]:
         """
-        Get data about all branches ("hoofdvestiging" + "nevenvestigingen") of a company
+        Get data about all branches ("hoofdvestiging" + "nevenvestigingen") of a company.
+
+        Filter response from KvK API (remove elements which are not specific branches) and
+        sort the results (move the main branch ("hoofdvestiging") to the front of the list)
         """
         kwargs.update({"kvkNummer": kvk})
-        return self.search(**kwargs).get("resultaten", [])
+        branches = self.search(**kwargs).get("resultaten", [])
+
+        for branch in branches:
```

Comment:
I'm generally wary of modifying lists that are being looped over, since this can lead to unexpected behaviour. For example, code like this runs infinitely because the list is modified during the loop:

```python
my_list = [1, 2, 3, 4, 5]

for item in my_list:
    print(item)
    if item == 2:
        my_list.insert(0, 6)
```

When testing with the test KvK API locally I didn't run into errors, but let's copy the branches list we loop over just to be sure.

```suggestion
        for branch in branches.copy():
```